build/resolver/git/entrypoint.sh

```diff
@@ -200,18 +213,30 @@ pull() {
         # encode each part of it and get '<encoded_user>:<encoded_password>'. If SCM_AUTH is in format '<token>',
         # give it a 'oauth2:' prefix to get 'oauth2:<encoded_token>'.
         if [ ! -z ${SCM_AUTH+x} ]; then
-            SCM_URL=$( modifyURL $SCM_URL $SCM_AUTH )
+            SCM_URL_MODIFIED=$( modifyURL $SCM_URL $SCM_AUTH )
         fi
 
         if [[ "${SOURCE_BRANCH}" == "${TARGET_BRANCH}" ]]; then
             echo "Clone $SOURCE_BRANCH..."
-            git clone -v -b master ${GIT_DEPTH_OPTION:-} --single-branch --recursive ${SCM_URL} data
+            git clone -v -b master ${GIT_DEPTH_OPTION:-} --single-branch --recursive ${SCM_URL_MODIFIED} data 2>&1 | tee clone_result.log 
+            if grep "The requested URL returned error: 500" clone_result.log > /dev/null 2>&1; then
```

Comment:
Make this code block `if ..... fi` to a common func?

Reply:
done